packages/renderer/src/lib/preferences/PreferencesCliTool.svelte

```diff
@@ -0,0 +1,135 @@
+<script lang="ts">
+import { faCircleArrowUp, faCircleXmark } from '@fortawesome/free-solid-svg-icons';
+import Button from '../ui/Button.svelte';
+import Fa from 'svelte-fa';
+import { startTask, type ConnectionCallback, eventCollect } from './preferences-connection-rendering-task';
+import type { CliToolInfo } from '../../../../main/src/plugin/api/cli-tool-info';
+import LoadingIconButton from '../ui/LoadingIconButton.svelte';
+
+import Markdown from '../markdown/Markdown.svelte';
+import type { ILoadingStatus } from './Util';
+
+export let cliTool: CliToolInfo;
+let showError = false;
+let cliToolStatus: ILoadingStatus = {
+  inProgress: false,
+  status: cliTool.newVersion ? 'toUpdate' : 'unknown',
+  action: 'update',
+};
+
+async function update(cliTool: CliToolInfo) {
+  try {
+    cliToolStatus.inProgress = true;
+    cliToolStatus = cliToolStatus;
+    const loggerHandlerKey = startTask(
+      `Update ${cliTool.name} to v${cliTool.newVersion}`,
+      '/preferences/cli-tools',
+      getLoggerHandler(cliTool.id),
+    );
+    await window.updateCliTool(cliTool.id, loggerHandlerKey, eventCollect);
+    showError = false;
+    cliToolStatus.status = 'unknown';
+  } catch (e) {
+    showError = true;
+  } finally {
+    cliToolStatus.inProgress = false;
+    cliToolStatus = cliToolStatus;
+  }
+}
+
+function getLoggerHandler(_cliToolId: string): ConnectionCallback {
+  return {
+    log: () => {},
+    warn: () => {},
```

Comment:
should we be able to see these logs in case of errors ?

Reply:
No idea. More a question for @mairin . Maybe it's enough the task manager?

Reply:
But i can enhance it on a separate PR so Jeff can work on the kubectl if it must be ready for the release